test/ml_dsa_test.c

```diff
@@ -493,6 +493,32 @@ static int ml_dsa_digest_sign_verify_test(void)
     return ret;
 }
 
+static int ml_dsa_priv_pub_bad_t0_test(void)
+{
+    int ret = 0;
+    EVP_PKEY *key = NULL;
+    ML_DSA_SIG_GEN_TEST_DATA *td = &ml_dsa_siggen_testdata[0];
+    uint8_t *priv = OPENSSL_memdup(td->priv, td->priv_len);
+
+    if (!TEST_ptr(priv))
+        goto err;
+    memcpy(priv, td->priv, td->priv_len);
+    /*
+     * t0 is at the end of the encoding so corrupt it.
+     * This offset is the start of t0 (which is the last 416 * k bytes))
+     */
+    priv[td->priv_len - 6 * 416] ^= 1;
+
+    if (!TEST_true(ml_dsa_create_keypair(&key, td->alg,
```

Comment:
I don't understand. Why `TEST_true`? Surely the point of this test is that we expect this to fail?

Reply:
This calls a function in which the last parameter passed is "expected" which is 0..
The function has multiple calls and I test for the4 expected failure point, A general failure anywhere in the call is not sufficient.
So the test is testing that it is true that it fails as expected (evp_test does something similar using "Result")

Reply:
changed it to use a variable to make it more obvious